### PR TITLE
DCOS_LAUNCH_ROLE_ARN specifies CF service role.

### PR DESCRIPTION
**Pull request by jdef**
## High-level description (required)

What features does this change enable? / What bugs does this change fix?

Unless RoleARN is specified, AWS CloudFormation will use a temporary
session w/ a fixed (limited) timeout. By specifying a RoleARN via the
DCOS_LAUNCH_ROLE_ARN envvar the timeout of the CF session may be
extended by adjusting a timeout parameter on the IAM role itself.

## Corresponding tickets (required)

These JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS-43800](https://jira.mesosphere.com/browse/DCOS-43800) The following resource(s) failed to delete: [BareServerAutoScale].

## Checklist for all PRs

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
    * unclear how to write a meaningful test that takes less than 60m. I tested this functionality w/ an internal project and it seems to be working as expected.
  - [x] **Added or updated any relevant documentation**

[Integration tests](https://teamcity.mesosphere.io/project.html?projectId=DcosIo_DcosLaunch&branch_DcosIo_DcosLaunch=%3Cdefault%3E) were run and

  - [ ] AWS Cloudformation Simple (link to job: )
  - [ ] Azure Resource Manager (link to job: )
  - [x] Onprem-AWS (link to job: )
  - [ ] Onprem-GCE (link to job: )

**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs should have two approved [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner.

### CODE REVIEW

**Comment by jdef** [comment]
@margaret @cprovencher PTAL

File: dcos_launch/platforms/aws.py [hunk 1]
```diff
@@ -164,6 +165,7 @@ def create_stack(
         Starts stack creation if validation is successful
         """
         log.info('Requesting AWS CloudFormation: {}'.format(name))
+        role_arn = os.getenv('DCOS_LAUNCH_ROLE_ARN', None)
```
Comment (cprovencher) [inline dcos_launch/platforms/aws.py]: nit: getenv "None" doesn't need to be specified because getenv returns None as a default

Comment (jdef) [reply dcos_launch/platforms/aws.py]: done

**Comment by cprovencher** [comment]
@jdef  I am off tomorrow, so I will merge this when I come in on Tuesday in case this brakes DC/OS CI.

**Comment by cprovencher** [comment]
@jdef  I am off tomorrow, so I will merge this when I come in on Tuesday in case this brakes DC/OS CI.

**Comment by cprovencher** [comment]
@jdef  I am off tomorrow, so I will merge this when I come in on Tuesday in case this brakes DC/OS CI.

**Pull request merged**